### PR TITLE
1260 CFG introduce analyzers

### PR DESCRIPTION
**Pull request by fm-117**
This PR focuses on the introduction of Analyzer concept. This is a first step in CFG integration but it is not complete as the new CfgBuildingMode option is not used yet and the CFG is still activated through `NodeDispatcher.RegisterStaticNodeListenerFactory`.

Next steps of CFG integration are :
- Completely remove `NodeDispatcher` class and replace static node listeners with analyzers
- Update all `CompilationProject` constructor usages to provide appropriate analyzers in each context

### CODE REVIEW

File: TypeCobol/Compiler/CupParser/NodeBuilder/ProgramClassBuilderNodeListener.cs [hunk 1]
```diff
@@ -27,14 +22,15 @@ public virtual void OnLevel1Definition(DataDefinition level1Node)
...
+
         }
 
         public virtual void Exit(Node node)
         {
 
         }
 
-        public virtual void AddNextSentenceStatement([NotNull] NextSentenceStatement stmt)
+        public virtual void AddNextSentenceStatement(NextSentenceStatement stmt)
```
Comment (smedilol) [inline TypeCobol/Compiler/CupParser/NodeBuilder/ProgramClassBuilderNodeListener.cs:33]: Why did you remove [NotNull] ?

Comment (fm-117) [reply TypeCobol/Compiler/CupParser/NodeBuilder/ProgramClassBuilderNodeListener.cs:33]: The `[NotNull]` attribute is set at interface level and is redundant here (ReSharper warning).

Comment (smedilol) [reply TypeCobol/Compiler/CupParser/NodeBuilder/ProgramClassBuilderNodeListener.cs:33]: Ok

**Comment by fm-117** [comment]
dummy comment to force PR refresh.

**Comment by fm-117** [comment]
PR misses a commit (Github downtime). To be closed and recreated.
